### PR TITLE
publish to terraform registry

### PR DESCRIPTION
**Pull request by ryanking**
Publish this provider to the terraform registry.

In terraform 0.13 third party providers can be published to the terraform registry. This PR implements the necessary pieces, including signing all releases.

## Test Plan
* [x] pushed 0.13.0+b2a3596 to the registry at https://registry.terraform.io/providers/chanzuckerberg/snowflake/latest?pollNotifications=true

## References
* https://docs.google.com/document/d/1J4p5KFH129wZbSF0XUioDbHSB7uZA-l2o1psLa-3YS4/edit

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,9 +38,9 @@ release: ## run a release
 release-prerelease: build ## release to github as a 'pre-release'
 	version=`./$(BASE_BINARY_NAME) -version`; \
 	git tag v"$$version"; \
-	git push
+	# git push
 	git push --tags
-	goreleaser release -f .goreleaser.prerelease.yml --debug
+	goreleaser release -f .goreleaser.prerelease.yml --debug --rm-dist
```
Comment (edulop91) [inline Makefile:43]: should this fail if KEYBASE_KEY_ID not set?

Comment (ryanking) [reply Makefile:43]: Yeah good idea

File: .goreleaser.yml [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
+    name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
 
-release:
-  prerelease: false
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
+  algorithm: sha256
 
-env_files:
-  github_token: ~/.config/goreleaser/github_token
```
Comment (edulop91) [inline .goreleaser.yml:25]: are these needed?

File: Makefile [hunk 3]
```diff
@@ -29,18 +28,24 @@ lint-all: fmt ## run the fast go linters
...
 	git push
 	goreleaser release
 .PHONY: release
 
-release-prerelease: build ## release to github as a 'pre-release'
+release-prerelease: check-release-prereqs build ## release to github as a 'pre-release'
 	version=`./$(BASE_BINARY_NAME) -version`; \
 	git tag v"$$version"; \
-	git push
+	# git push
```
Comment (edulop91) [inline Makefile:46]:
```suggestion
```

File: .goreleaser.yml (hunk 2 above)
Comment (ryanking) [reply .goreleaser.yml:25]: I have no idea of their purpose. Looks like you added it [here](https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/103)

Comment (ryanking) [reply .goreleaser.yml:25]: I have alwayst just had a token exported in my environment though.

**Pull request merged**